data/helpers.d/utils

```diff
@@ -685,3 +699,37 @@ ynh_compare_current_package_version() {
     # Return the return value of dpkg --compare-versions
     dpkg --compare-versions $current_version $comparison $version
 }
+
+# Check if we should enforce sane default permissions (= disable rwx for 'others')
+# on file/folders handled with ynh_setup_source and ynh_add_config
+#
+# [internal]
+#
+# Having a file others-readable or a folder others-executable(=enterable)
+# is a security risk comparable to "chmod 777"
+#
+# Configuration files may contain secrets. Or even just being able to enter a
+# folder may allow an attacker to do nasty stuff (maybe a file or subfolder has
+# some write permission enabled for 'other' and the attacker may edit the
+# content or create files as leverage for priviledge escalation ...)
+#
+# The sane default should be to set ownership to $app:$app.
+# In specific case, you may want to set the ownership to $app:www-data
+# for example if nginx needs access to static files.
+#
+_ynh_apply_default_permissions() {
+    local target=$1
+
+    local ynh_requirement=$(jq -r '.requirements.yunohost' $YNH_APP_BASEDIR/manifest.json)
+
+    if [ -z "$ynh_requirements" ] || [ "$ynh_requirements" == "null" ] || dpkg --compare-versions $ynh_requirements ge 4.2
+    then
+        chmod o-rwx $target
+        if ynh_system_user_exists $app
+        then
+            chown $app:$app $target
```

Comment:
chown -R www-data:$app

Reply:
Thinking again these last day about the initial discussion about this : this just sounds weird. If the point is to prevent that `$app` edit its own files, then let's set root:root as the owner recursively, then set `$app` as owner of `$target` (the point being that being owner provides r/x rights since we used `o-rwx`)